### PR TITLE
Annotation models

### LINKED ISSUE #23: Update schema to support annotations
Implement the annotation tables from the schema diagram. There are a lot of tables for annotations, so this will actually be a decently involved task.

Tables:
 - Publication
 - AnnotationStatus
 - Annotation
 - GeneTermAnnotation
 - GeneGeneAnnotation
 - CommentAnnotation

Acceptance Criteria:
- The above list of tables is implemented
- Tests proving they actually work

### PR DESCRIPTION
**Pull request by JDonut**
Closes #23

### CODE REVIEW

**Comment by JDonut** [comment]
There are a couple TODOs in there referencing Locus ID lookup. We don't currently know exactly how we're going to store Locuses yet. We will return to these TODOs once that has been figured out.

File: app/models/annotation.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+require('./publication');
+require('./annotation_status');
+require('./user');
+require('./keyword');
+require('./gene_term_annotation');
+require('./gene_gene_annotation');
+require('./comment_annotation');
+
+let Annotation = bookshelf.model('Annotation', {
```
Comment (jcdesimp) [inline app/models/annotation.js]: Minor thing but this could probably be a const, up to you.

File: app/models/annotation_status.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+"use strict";
+
+const bookshelf = require('../lib/bookshelf');
+
+require('./annotation');
+
+let AnnotationStatus = bookshelf.model('AnnotationStatus', {
```
Comment (jcdesimp) [inline app/models/annotation_status.js]: same here...

File: app/models/comment_annotation.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+"use strict";
+
+const bookshelf = require('../lib/bookshelf');
+
+require('./annotation');
+
+let CommentAnnotation = bookshelf.model('CommentAnnotation', {
```
Comment (jcdesimp) [inline app/models/comment_annotation.js]: and again...

File: app/models/gene_gene_annotation.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+"use strict";
+
+const bookshelf = require('../lib/bookshelf');
+
+require('./keyword');
+require('./annotation');
+
+let GeneGeneAnnotation = bookshelf.model('GeneGeneAnnotation', {
```
Comment (jcdesimp) [inline app/models/gene_gene_annotation.js]: can use const if you want...

File: app/models/gene_term_annotation.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const bookshelf = require('../lib/bookshelf');
+
+require('./keyword');
+require('./annotation');
+
+let GeneTermAnnotation = bookshelf.model('GeneTermAnnotation', {
```
Comment (jcdesimp) [inline app/models/gene_term_annotation.js]: same...

File: app/models/publication.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+"use strict";
+
+const bookshelf = require('../lib/bookshelf');
+
+require('./annotation');
+
+let Publication = bookshelf.model('Publication', {
```
Comment (jcdesimp) [inline app/models/publication.js]: same...

File: test/models_test.js [hunk 7]
```diff
@@ -88,4 +95,230 @@ describe('Models', function() {
...
+					let actual = res.toJSON();
+					chai.expect(actual.referencedBy).to.containSubset(expectedAnnotations);
+					done();
+				});
+		});
+
+	});
+
+	describe('Annotation Status', function() {
+
```
Comment (jcdesimp) [inline test/models_test.js]: some extra whitespace in this file can be cleaned up if you so choose.

**Pull request merged**